src/TTextEdit.cpp

```diff
@@ -1784,6 +1805,433 @@ int TTextEdit::getRowCount()
     return height() / rowHeight;
 }
 
+inline QString TTextEdit::htmlCenter(const QString& text)
+{
+    return QStringLiteral("<center>%1</center>").arg(text);
+}
+
+// Not just whitespace but also some formatting and other things - it may be
+// that some entries do not work like this and we cannot just display a short
+// bit of text to indicate them in the analysis of the on-screen content- the
+// language directional controls may be like that:
+inline QString TTextEdit::convertWhiteSpaceToVisual(const QChar& first, const QChar& second)
+{
+    // clang-format off
+    if (second.isNull()) {
+        // The code point is on the BMP
+        switch (first.unicode()) {
+        case QChar::Tabulation:         return htmlCenter(tr("{tab}"));     break;
+        case QChar::LineFeed:           return htmlCenter(tr("{lf}"));      break; // Not likely to be seen as it gets filtered out
+        case QChar::CarriageReturn:     return htmlCenter(tr("{cr}"));      break; // Not likely to be seen as it gets filtered out
+        case QChar::Space:              return htmlCenter(tr("{sp}"));      break;
+        case QChar::Nbsp:               return htmlCenter(tr("{nbsp}"));    break;
+        case QChar::SoftHyphen:         return htmlCenter(tr("{shyp}"));    break;
+        case 0x034F:                    return htmlCenter(tr("{cgj}"));     break; // Combining Grapheme Joiner (badly named apparently)
+        case 0x1680:                    return htmlCenter(tr("{osp}"));     break; // Ogham space mark
+        case 0x2000:                    return htmlCenter(tr("{enqd}"));    break; // En quad
+        case 0x2001:                    return htmlCenter(tr("{emqd}"));    break; // Em quad
+        case 0x2002:                    return htmlCenter(tr("{ensp}"));    break; // En space
+        case 0x2003:                    return htmlCenter(tr("{emsp}"));    break; // Em space
+        case 0x2004:                    return htmlCenter(tr("{m3sp}"));    break; // Three-per-em (Thick) space
```

Comment:
How would our users know what {m3sp} means? Shouldn't the comment be the text?

This also isn't something that should be translated, is it?

Reply:
How can we expect our users to know what it means? I think we should put the full names in instead. It's unlikely there would be entire lines composed of only these characters, right?

Reply:
I fear there is a risk that there will not be enough space to display everything on-screen, a single word might be okay, but a phrase as *some* of them are will be problematic IMHO.

Reply:
Okay!